### PR TITLE
Added reshape & drag functionality to bounding boxes

### PR DESCRIPTION
**Pull request by willalexander**

### CODE REVIEW

File: resources/js/components/Admin/Box.vue [hunk 1]
```diff
@@ -19,121 +19,97 @@
...
+	    dragEnd()
+	    {
+		this.$emit('dragEnd');
+	    }
         }
     }
 </script>
 
 <style lang="scss" scoped>
-
```
Comment (xlcrr) [inline resources/js/components/Admin/Box.vue:77]: Please keep these spaces, they were added intentionally

File: resources/js/views/admin/BoundingBox.vue [hunk 2]
```diff
@@ -81,167 +74,176 @@ export default {
...
+	    activatedNode: 0,
+	    currentX: 0,
+	    currentY: 0,
+	    dir: [[0, 1],[1, 0],[0, 1],[1, 0]],
+	    apply: [[1, 0, 0, -1],[0, 1, -1, 0],[0, 0, 0, 1],[0, 0, 1, 0]], 
+	    dragBox: -1
         };
     },
     computed: {
-
```
Comment (xlcrr) [inline resources/js/views/admin/BoundingBox.vue:93]: Please keep these spaces, they were added intentionally

File: resources/js/components/Admin/Box.vue [hunk 3]
```diff
@@ -101,7 +93,6 @@
     .top-node {
```
Comment (lhtrung307) [inline resources/js/components/Admin/Box.vue:93]: Should refactor css for 4 nodes

File: resources/js/views/admin/BoundingBox.vue [hunk 4]
```diff
@@ -11,35 +11,30 @@
...
                             :activeTop="box.activeTop"
                             :activeLeft="box.activeLeft"
                             :activeBottom="box.activeBottom"
                             :activeRight="box.activeRight"
-                            @select="selectBox(i)"
+                            @select="selectBox"
                             @activate="activate"
                             @deselectNode="deselectNode"
                             @repositionTop="repositionTop"
+			    @dragEnd="dragEnd"
```
Comment (lhtrung307) [inline resources/js/views/admin/BoundingBox.vue:37]: Where did `dragEnd` define?

**Comment by lhtrung307** [comment]
Should have a function that allows admin to delete box.

File: resources/js/views/admin/BoundingBox.vue (hunk 4 above)
Comment (willalexander) [reply resources/js/views/admin/BoundingBox.vue:37]: It's defined in the methods section, line 248.

The box child component emits a 'dragEnd' event when the user lifts the mouse button

File: resources/js/components/Admin/Box.vue (hunk 3 above)
Comment (willalexander) [reply resources/js/components/Admin/Box.vue:93]: Should I reduce them to a single class, as most of the properties are duplicated? Something like this:

.node {
        position: absolute;
        height: 10px;
        width: 10px;
        background-color: #90ee90;
        cursor: grab;
    }

Comment (lhtrung307) [reply resources/js/components/Admin/Box.vue:93]: Yes :D.

Comment (willalexander) [reply resources/js/components/Admin/Box.vue]: done!

**Comment by lhtrung307** [comment]
If I drag the bottom of the box to above of top, the box border is disappear.

**Comment by willalexander** [comment]
> If I drag the bottom of the box to above of top, the box border is disappear.

Thanks for catching this! I've fixed it in my latest commit

File: resources/js/components/Admin/Box.vue [hunk 5]
```diff
@@ -1,8 +1,8 @@
 <template>
-    <div class="box-wrapper" @mousedown="function(e) { selectBox(e) }" @mouseup="dragEnd()">
+    <div class="box-wrapper" @mousedown="function(e) { selectBox(e) }" @mouseup="mouseUp()">
 
-        <div :class="selected ? 'box selected-box' : 'box'"
-             :style="{top: this.geom[0] + 'px', left: this.geom[1] + 'px', width: this.geom[2] + 'px', height: this.geom[3] + 'px'}">
+        <div :class="selected ? 'box selected-box' : 'box'
+"             :style="{top: this.geom[0] + 'px', left: this.geom[1] + 'px', width: this.geom[2] + 'px', height: this.geom[3] + 'px'}">
```
Comment (lhtrung307) [inline resources/js/components/Admin/Box.vue:5]: I think this double-quote should stay in the same row with class. If you were familiar with eslint, I recommend pulling new code from master and let's eslint do its job 😄.

File: resources/js/views/admin/BoundingBox.vue [hunk 6]
```diff
@@ -153,27 +158,43 @@ export default {
...
+                var newWidth = this.startGeom[2] + this.apply[this.activatedNode][2] * diff;
+                var newHeight = this.startGeom[3] + this.apply[this.activatedNode][3] * diff;
 
-                this.boxes[this.activatedBox].geom = [
-                    this.boxes[this.activatedBox].geom[0] + this.apply[this.activatedNode][0] * diff,
-                    this.boxes[this.activatedBox].geom[1] + this.apply[this.activatedNode][1] * diff,
-                    this.boxes[this.activatedBox].geom[2] + this.apply[this.activatedNode][2] * diff,
-                    this.boxes[this.activatedBox].geom[3] + this.apply[this.activatedNode][3] * diff
-                ];
+                if((newWidth > 43)&&(newHeight > 43))
```
Comment (lhtrung307) [inline resources/js/views/admin/BoundingBox.vue:189]: Should not use magic number.

**Comment by xlcrr** [comment]
It's very smooth. This is great work.

On web.php, change the route for `bbox` to this and you will be able to visit the page easier 

```
 // Add coordinates
 Route::get('bbox', 'HomeController@index');`
```

This will redirect to vue SPA and the frontend routes.js will handle the redirect once you have admin credentials

One problem is that if I have 2 boxes and click delete, both are deleted.

However, I am happy to merge this into production and we can branch off it again

**Review by xlcrr — APPROVED** [review]

File: resources/js/components/Admin/Box.vue (hunk 5 above)
Comment (willalexander) [reply resources/js/components/Admin/Box.vue:5]: Thanks! I will take a look at eslint.

File: resources/js/views/admin/BoundingBox.vue (hunk 6 above)
Comment (willalexander) [reply resources/js/views/admin/BoundingBox.vue:189]: ok, I will replace that with a constant

**Comment by willalexander** [comment]
> It's very smooth. This is great work.
> 
> On web.php, change the route for `bbox` to this and you will be able to visit the page easier
> 
> ```
>  // Add coordinates
>  Route::get('bbox', 'HomeController@index');`
> ```
> 
> This will redirect to vue SPA and the frontend routes.js will handle the redirect once you have admin credentials
> 
> One problem is that if I have 2 boxes and click delete, both are deleted.
> 
> However, I am happy to merge this into production and we can branch off it again

Thank you!

And thanks for the tip about changing the web.php route, I was having to hack it until now..

I'll fix the outstanding issues that you & lhtrung307 have raised.

**Pull request merged**